prow/cmd/build/controller_test.go

```diff
@@ -214,6 +291,385 @@ func TestEnqueueKey(t *testing.T) {
 	}
 }
 
+func TestTerminateDupProwJobs(t *testing.T) {
```

Comment:
This seems like the wrong level of test for this package -- the filtering of jobs to determine dupes and the actual termination are library code, can we mock those and test that the full set of duped jobs as determined by the library get the termination called on them? Then test that the correct set of jobs gets dupes identified in a test for the library, as well as that the termination works as expected in the library.

An integration test with a set of fake clients that checks for ProwJob delete calls also good for this package

Reply:
I added it here in order to cover the functionality end to end. Also this is on par with the existing plank tests: https://github.com/kubernetes/test-infra/blob/9c83e851fd7bae91648585b3a7d6897d67cf08b5/prow/plank/controller_test.go#L197

Reply:
I could add also a dedicated test in the aborter, but I think that would be already covered by this one.

Reply:
The existing plank tests are at that level because there was no library when they were written :)

We need to test the functionality of the aborter in the aborter package, and test that given a functional aborter we do what we should in here. An integration test would be useful as well.

Reply:
I added a separated tests for aborter.